src/lsstseries/analysis/structure_function/base_calculator.py

```diff
@@ -84,6 +86,80 @@ def _bin_dts(self, dts):
 
+    def _calculate_binned_statistics(self, statistic_to_apply="mean"):
+        """This method will take the parallel delta_t and delta_flux arrays,
+        bin the delta_t values using the bin edges defined by self._bins. Then
+        the corresponding delta_flux values in each bin will have a statistic
+        measure applied.
+
+        Largely speaking this is a wrapper over Scipy's `binned_statistic`, so
+        any of the statistics supported by that function are valid inputs here.
+
+        Parameters
+        ----------
+        statistict_to_apply : str or function, optional
+            The statistic to apply to the values in each delta_t bin, by default
+            "mean".
+
+        Returns
+        -------
+        Two List[floats]
+            The first list returned defines the center of the delta_t bins.
+            The second list returned contains the result of evaluating the
+            statistic measure of the delta_flux values in each delta_t bin.
+        """
+
+        # combining treats all lightcurves as one when calculating the structure function
+        if self._argument_container.combine and len(self._time) > 1:
+            self._dts = np.hstack(np.array(self._dts, dtype="object"))
+            self._all_d_fluxes = np.hstack(np.array(self._all_d_fluxes, dtype="object"))
+
+            # binning
+            if self._bins is None:
+                self._bin_dts(self._dts)
+
+            # structure function at specific dt
+            # the line below will throw error if the bins are not covering the whole range
+            sfs, bin_edgs, _ = binned_statistic(
+                self._dts, self._all_d_fluxes, statistic=statistic_to_apply, bins=self._bins
+            )
+
+            # return the mean delta_time values for each bin
+            # bin_means, _, _ = binned_statistic(self._dts, self._dts, statistic="mean", bins=self._bins)
+            return [(bin_edgs[0:-1] + bin_edgs[1:]) / 2], [sfs]
+
+        # Not combining calculates structure function for each light curve independently
+        else:
+            # may want to raise warning if len(times) <=1 and combine was set true
+            sfs_all = []
+            t_all = []
+            for lc_idx in range(len(self._dts)):
+                if len(self._dts[lc_idx]) > 1:
+                    # bin the delta_time values, and evaluate the `statistic_to_apply`
+                    # for the delta_flux values in each bin.
+                    self._bin_dts(self._dts[lc_idx])
+                    sfs, bin_edgs, _ = binned_statistic(
+                        self._dts[lc_idx],
+                        self._all_d_fluxes[lc_idx],
+                        statistic=statistic_to_apply,
+                        bins=self._bins,
+                    )
+                    sfs_all.append(sfs)
+
+                    # return the mean delta_time values for each bin
+                    # bin_means, _, _ = binned_statistic(
```

Comment:
This commented out line is a place holder for a future PR. I didn't want to uncomment it here, because when it is used, it will require making an adjustment to almost every unit test.

Separating that from this bit of abstraction work seemed like a good idea, so that we can see that this change doesn't affect any of the unit tests.